core/src/main/java/org/infinispan/interceptors/distribution/TxDistributionInterceptor.java

```diff
@@ -305,7 +345,7 @@ private Object localGet(InvocationContext ctx, Object key, boolean isWrite,
          }
          if (!ctx.replaceValue(key, ice)) {
             if (isWrite)
-               lockAndWrap(ctx, key, ice, command);
+               entryFactory.wrapEntryForPut(ctx, key, ice, false, command, false);
```

Comment:
I don't see why it is needed for the first place... in optimistic, the prepare will acquire the locks. In pessimistic, the lock is acquired via ```LockControlCommand```. In any case, if a topology happens, the state transfer will exchange this information anyway.